### DISCUSSION TITLE
Survival Probability plots for RSF Survival function

### DISCUSSION THREAD

**Discussion by nkaur201** [sebp/scikit-survival#286] [Q&A]
Hello, 

At the start of the tutorial an example of the Kaplan meier plots is given where patient groups (treatment) are compared. 

Are we able to do the same but using the random forest survival predict function instead? 

Instead of comparing individual patients, I would like to also group them. 

If so, how would we implement this. 

Thank you!

![image](https://user-images.githubusercontent.com/36765835/183864059-27ed91cb-18f7-4a0f-bf19-30ff2980218c.png)

**Comment by sebp** [comment]
A random survival forest always predicts a survival function per sample (a row in `X`), but cannot predict a single survival function for a set of samples. If you have a binary (or categorical) feature, you can of course manipulate the `X` passed to `predict_survival_function` to obtain 2 curves. Let's say the second column in `X` corresponds to treatment, which can take on values 0 and 1, then you can do:

```python
X_0 = X[[11], :].copy()  # pick a patient 11
X_0[:, 1] = 0  # set treatment to 0
surv_0 = rsf.predict_survival_function(X_0)

X_1 = X[[11], :].copy()  # pick a patient 11
X_1[:, 1] = 1  # set treatment to 1
surv_1 = rsf.predict_survival_function(X_1)
```

The you can compare the survival function for treatment and no treatment with everything else being equal.

**Reply by nkaur201** [reply]
Thank you! Appreciate this!
